**Hands-on Activity 7.1 : Classifications and Regression**

Name: **Espiritu, Diego Angelo G.**

Course and Section: **CPE019 - CPE32S1**

Instructor: **Engr. Roman Richard**

Date: **06/30/2024**

1. Choose any dataset applicable to the classification problem, and also, choose any dataset applicable to the regression problem.
2. Explain your datasets and the problem being addressed.
3. For classification, do the following:
  1. Create a base model
  2. Evaluate the model with k-fold cross validation
  3. Improve the accuracy of your model by applying additional hidden layers
4. For regression, do the following:
  1. Create a base model
  2. Improve the model by standardizing the dataset
  3. Show tuning of layers and neurons (see evaluating small and larger networks)
5. Submit the link to your Google Colab (make sure that it is accessible to me)

In [ ]:
import pandas as pd
import numpy as np

housingDF = pd.read_csv("/content/drive/MyDrive/CPE019/Housing.csv") #Regression type dataset
appleDF = pd.read_csv("/content/drive/MyDrive/CPE019/apple_quality.csv") #Classification type dataset

In [ ]:
appleDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4000 entries, 0 to 3999
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   A_id         4000 non-null   int64  
 1   Size         4000 non-null   float64
 2   Weight       4000 non-null   float64
 3   Sweetness    4000 non-null   float64
 4   Crunchiness  4000 non-null   float64
 5   Juiciness    4000 non-null   float64
 6   Ripeness     4000 non-null   float64
 7   Acidity      4000 non-null   float64
 8   Quality      4000 non-null   object 
dtypes: float64(7), int64(1), object(1)
memory usage: 281.4+ KB


The Apple Quality dataset is the classification dataset that was used where classifiers more especially, binary classifiers are the target variable and the quality variable is limited to two labels good and bad. Developing a model for binary classification data types based on the input variables included in the dataset is the challenge with this dataset.

In [ ]:
housingDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

The Housing Prices dataset is the one used for regression analysis. wherein continuous data more especially, the price itself is the target variable. Finding and developing a model for regression type data based on the input variables included in the dataset is the challenge with this dataset.

# Classification Problem

In [ ]:
from keras.models import Sequential
from keras.layers import Dense, Flatten
from scikeras.wrappers import KerasClassifier
from tensorflow.keras.utils import to_categorical
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline

In [ ]:
# Size: Size of the fruit
# Weight: Weight of the fruit
# Sweetness: Degree of sweetness of the fruit
# Crunchiness: Texture indicating the crunchiness of the fruit
# Juiciness: Level of juiciness of the fruit
# Ripeness: Stage of ripeness of the fruit
# Acidity: Acidity level of the fruit
# Quality: Overall quality of the fruit
columns = ["Size", "Weight", "Sweetness", "Crunchiness", "Juiciness", "Ripeness", "Acidity"]
apple_x = appleDF[list(columns)].values
apple_y = appleDF["Quality"].values

In [ ]:
# encode class values as integers
seed = 7
np.random.seed(seed)
encoder = LabelEncoder()
encoder.fit(apple_y)
encoded_Y = encoder.transform(apple_y)
# convert integers to dummy variables (i.e. one hot encoded)
dummy_y = to_categorical(encoded_Y)

In [ ]:
# define baseline model
def baseline_model():
  # create model
  model = Sequential()
  model.add(Dense(60, input_shape = (len(columns),), activation= 'relu' ))
  model.add(Dense(1, activation= 'sigmoid' ))
  # Compile model
  model.compile(loss= 'binary_crossentropy' , optimizer= 'adam' , metrics=[ 'accuracy' ])
  return model

In [ ]:
# Evaluate The Model with k-Fold Cross Validation
estimator = KerasClassifier(model=baseline_model, epochs=100, batch_size=5, verbose = 1)
kfold = KFold(n_splits=10, shuffle=True, random_state=seed)
results = cross_val_score(estimator, apple_x, dummy_y, cv=kfold)
print("Accuracy: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))

Epoch 1/100


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.5027 - loss: 0.7967
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.6851 - loss: 0.6174
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7705 - loss: 0.5223
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7870 - loss: 0.4853
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8106 - loss: 0.4516
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8146 - loss: 0.4289
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8096 - loss: 0.4289
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7999 - loss: 0.4337
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8177 - loss: 0.4131
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8211 - loss: 0.4030
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8224 - loss: 0.4138
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.4557 - loss: 1.1653
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7433 - loss: 0.5401
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7644 - loss: 0.4931
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7830 - loss: 0.4699
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7987 - loss: 0.4559
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.8012 - loss: 0.4326
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7963 - loss: 0.4431
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8113 - loss: 0.4157
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8098 - loss: 0.4283
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8146 - loss: 0.4160
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8103 - loss: 0.4130
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.6742 - loss: 0.6908
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7759 - loss: 0.4614
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7861 - loss: 0.4470
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8064 - loss: 0.4143
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7889 - loss: 0.4340
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8074 - loss: 0.4160
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8046 - loss: 0.4037
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8218 - loss: 0.3916
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8169 - loss: 0.4064
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8229 - loss: 0.3780
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8205 - loss: 0.3846
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.6374 - loss: 0.7019
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7550 - loss: 0.5008
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7926 - loss: 0.4329
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8104 - loss: 0.4162
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8320 - loss: 0.3820
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8174 - loss: 0.3956
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8279 - loss: 0.3658
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8347 - loss: 0.3746
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8344 - loss: 0.3716
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8175 - loss: 0.3899
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8384 - loss: 0.3588
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 54s 2ms/step - accuracy: 0.3977 - loss: 1.5290
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6145 - loss: 0.6418
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7689 - loss: 0.4967
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7746 - loss: 0.4562
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7957 - loss: 0.4249
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8031 - loss: 0.4075
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8175 - loss: 0.3979
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8196 - loss: 0.3815
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8253 - loss: 0.3765
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8145 - loss: 0.3758
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8277 - loss: 0.3692
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/ste

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.5802 - loss: 0.7638
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7206 - loss: 0.5338
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7561 - loss: 0.4910
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7814 - loss: 0.4569
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7910 - loss: 0.4438
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8004 - loss: 0.4198
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7983 - loss: 0.4126
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7895 - loss: 0.4242
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8175 - loss: 0.3973
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8031 - loss: 0.4015
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8147 - loss: 0.3920
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.6211 - loss: 0.6673
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7442 - loss: 0.5025
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7728 - loss: 0.4710
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7863 - loss: 0.4447
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7781 - loss: 0.4301
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7976 - loss: 0.4192
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7836 - loss: 0.4309
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8057 - loss: 0.3919
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7956 - loss: 0.4200
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7946 - loss: 0.4154
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7975 - loss: 0.4052
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.5409 - loss: 0.8627
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7173 - loss: 0.5472
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7694 - loss: 0.4989
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7854 - loss: 0.4723
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8035 - loss: 0.4452
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7957 - loss: 0.4389
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8009 - loss: 0.4382
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7917 - loss: 0.4497
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8089 - loss: 0.4266
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8083 - loss: 0.4361
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7955 - loss: 0.4482
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.4920 - loss: 0.7536
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 4s 4ms/step - accuracy: 0.6423 - loss: 0.6112
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7344 - loss: 0.5378
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7594 - loss: 0.5166
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7870 - loss: 0.4759
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7890 - loss: 0.4486
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8084 - loss: 0.4243
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8241 - loss: 0.3977
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8276 - loss: 0.3914
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8235 - loss: 0.3873
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.8174 - loss: 0.3889
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.3961 - loss: 1.4115
Epoch 2/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.4627 - loss: 0.7422
Epoch 3/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6208 - loss: 0.6575
Epoch 4/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.6940 - loss: 0.5946
Epoch 5/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.7587 - loss: 0.5273
Epoch 6/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7752 - loss: 0.4971
Epoch 7/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8054 - loss: 0.4599
Epoch 8/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8030 - loss: 0.4554
Epoch 9/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8088 - loss: 0.4323
Epoch 10/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8158 - loss: 0.4237
Epoch 11/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8207 - loss: 0.4225
Epoch 12/100
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

With a standard deviation of 2.01%, the accuracy of the baseline model is 83.42%. The model has been run ten times, and the KFold validation has assessed the outcomes of those ten runs. The model was trained for 20 minutes before finishing, and while its accuracy score of 83.42% may seem impressive, it appears to perform worse than neural networks when compared to other examples.

In [ ]:
# Improve the accuracy of your model by applying additional hidden layers
# define baseline model
def model_2():
  # create model
  model = Sequential()
  model.add(Dense(60, input_dim = len(columns), activation= 'relu' ))
  model.add(Dense(120, activation = 'relu'))
  model.add(Dense(2, activation= 'sigmoid' ))
  # Compile model
  model.compile(loss= 'binary_crossentropy' , optimizer= 'adam' , metrics=[ 'accuracy' ])
  return model

In [ ]:
# Evaluate The Model with k-Fold Cross Validation
estimator = KerasClassifier(model=model_2, epochs=20, batch_size=5, verbose = 1)
kfold = KFold(n_splits=10, shuffle=True, random_state=seed)
results = cross_val_score(estimator, apple_x, dummy_y, cv=kfold)
print("Accuracy: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


Epoch 1/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7861 - loss: 0.4616
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8749 - loss: 0.3000
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8931 - loss: 0.2516
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9111 - loss: 0.2350
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9148 - loss: 0.2221
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9172 - loss: 0.2061
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9254 - loss: 0.1829
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9094 - loss: 0.2078
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9360 - loss: 0.1674
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9410 - loss: 0.1651
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9363 - loss: 0.1615
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7550 - loss: 0.4735
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8652 - loss: 0.2994
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8832 - loss: 0.2785
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9006 - loss: 0.2491
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9236 - loss: 0.2121
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9089 - loss: 0.2200
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9243 - loss: 0.2001
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9233 - loss: 0.1862
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9266 - loss: 0.1850
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9289 - loss: 0.1797
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9368 - loss: 0.1596
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7844 - loss: 0.4555
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8819 - loss: 0.2799
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8890 - loss: 0.2632
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9127 - loss: 0.2322
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9204 - loss: 0.2026
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9200 - loss: 0.1951
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9138 - loss: 0.1895
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9282 - loss: 0.1837
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9298 - loss: 0.1686
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9368 - loss: 0.1551
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9478 - loss: 0.1462
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7879 - loss: 0.4523
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8667 - loss: 0.2955
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8911 - loss: 0.2469
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9015 - loss: 0.2311
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9101 - loss: 0.2041
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9281 - loss: 0.1864
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9181 - loss: 0.1864
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9308 - loss: 0.1822
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9286 - loss: 0.1743
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9358 - loss: 0.1605
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9448 - loss: 0.1449
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.8009 - loss: 0.4527
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8642 - loss: 0.3111
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8973 - loss: 0.2550
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8992 - loss: 0.2365
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9081 - loss: 0.2222
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9211 - loss: 0.1976
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9126 - loss: 0.1993
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9191 - loss: 0.1894
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9280 - loss: 0.1762
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9335 - loss: 0.1746
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9333 - loss: 0.1518
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8032 - loss: 0.4498
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8729 - loss: 0.2844
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9007 - loss: 0.2503
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9041 - loss: 0.2350
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9073 - loss: 0.2232
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9157 - loss: 0.1971
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9237 - loss: 0.1844
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9286 - loss: 0.1626
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9304 - loss: 0.1710
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9456 - loss: 0.1482
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9468 - loss: 0.1425
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7820 - loss: 0.4617
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8722 - loss: 0.2952
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8805 - loss: 0.2703
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9001 - loss: 0.2416
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9140 - loss: 0.2105
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9186 - loss: 0.2023
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9224 - loss: 0.1925
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9357 - loss: 0.1620
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9362 - loss: 0.1599
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9439 - loss: 0.1457
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9385 - loss: 0.1527
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7707 - loss: 0.4684
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8629 - loss: 0.3021
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8931 - loss: 0.2504
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9111 - loss: 0.2316
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9233 - loss: 0.1994
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9248 - loss: 0.1942
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9253 - loss: 0.1816
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9285 - loss: 0.1815
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9306 - loss: 0.1704
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9394 - loss: 0.1607
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9447 - loss: 0.1522
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.7784 - loss: 0.4578
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8683 - loss: 0.2952
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9001 - loss: 0.2545
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9022 - loss: 0.2305
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9189 - loss: 0.2112
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9235 - loss: 0.2086
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9234 - loss: 0.2040
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9295 - loss: 0.1815
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9361 - loss: 0.1794
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9368 - loss: 0.1665
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9401 - loss: 0.1598
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7900 - loss: 0.4524
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8738 - loss: 0.2934
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8985 - loss: 0.2440
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9113 - loss: 0.2246
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9091 - loss: 0.2146
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9217 - loss: 0.1868
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9305 - loss: 0.1792
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9362 - loss: 0.1660
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9469 - loss: 0.1341
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9502 - loss: 0.1382
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9466 - loss: 0.1414
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

By using extra hidden layers one layer with 120 nodes, specifically and fewer epochs per fold specifically, 20 instead of 100 the accuracy of the second model increases from the baseline 83.42% by 10% to **93.60%** accuracy with a low standard deviation of 1.41%

In [ ]:
# Improve the accuracy of your model by applying additional hidden layers
# define baseline model
def model_3():
  # create model
  model = Sequential()
  model.add(Dense(60, input_dim = len(columns), activation= 'relu' ))
  model.add(Dense(120, activation = 'relu'))
  model.add(Dense(120, activation = 'relu'))
  model.add(Dense(2, activation= 'sigmoid' ))
  # Compile model
  model.compile(loss= 'binary_crossentropy' , optimizer= 'adam' , metrics=[ 'accuracy' ])
  return model

In [ ]:
# Evaluate The Model with k-Fold Cross Validation with standardization
from sklearn.preprocessing import StandardScaler
estimators = []
estimators.append(('standardize', StandardScaler()))
estimators.append(('mlp', KerasClassifier(model=model_3, epochs=20, batch_size=5, verbose = 1)))
pipeline = Pipeline(estimators)
kfold = KFold(n_splits=10, shuffle=True, random_state=seed)
results = cross_val_score(estimator, apple_x, dummy_y, cv=kfold)
print("Accuracy: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))

Epoch 1/20


/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7782 - loss: 0.4500
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8845 - loss: 0.2726
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9046 - loss: 0.2348
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9052 - loss: 0.2280
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9156 - loss: 0.2033
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9225 - loss: 0.1900
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9296 - loss: 0.1750
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9295 - loss: 0.1722
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9414 - loss: 0.1680
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9429 - loss: 0.1471
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9485 - loss: 0.1403
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7669 - loss: 0.4557
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8864 - loss: 0.2750
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9141 - loss: 0.2216
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9099 - loss: 0.2218
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9257 - loss: 0.1835
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9314 - loss: 0.1733
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9408 - loss: 0.1535
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9497 - loss: 0.1380
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9492 - loss: 0.1329
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9538 - loss: 0.1185
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9594 - loss: 0.1098
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - accuracy: 0.7948 - loss: 0.4356
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8846 - loss: 0.2707
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8860 - loss: 0.2584
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9081 - loss: 0.2231
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9201 - loss: 0.2067
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9255 - loss: 0.1809
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9326 - loss: 0.1683
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9283 - loss: 0.1820
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9433 - loss: 0.1499
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9471 - loss: 0.1354
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9504 - loss: 0.1225
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7906 - loss: 0.4367
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8880 - loss: 0.2689
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9075 - loss: 0.2369
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9139 - loss: 0.2060
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9175 - loss: 0.1949
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9243 - loss: 0.1808
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9346 - loss: 0.1599
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9384 - loss: 0.1490
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9475 - loss: 0.1328
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9457 - loss: 0.1309
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9597 - loss: 0.1104
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7688 - loss: 0.4490
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8759 - loss: 0.2917
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9044 - loss: 0.2340
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9054 - loss: 0.2351
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9067 - loss: 0.2162
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9263 - loss: 0.1875
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 3ms/step - accuracy: 0.9408 - loss: 0.1527
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9408 - loss: 0.1458
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9501 - loss: 0.1353
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9558 - loss: 0.1288
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9617 - loss: 0.1042
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7964 - loss: 0.4350
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.8884 - loss: 0.2502
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9009 - loss: 0.2337
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9127 - loss: 0.2176
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9204 - loss: 0.1884
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9387 - loss: 0.1590
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9407 - loss: 0.1406
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9496 - loss: 0.1470
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 3ms/step - accuracy: 0.9604 - loss: 0.1088
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9558 - loss: 0.1155
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9578 - loss: 0.1153
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7809 - loss: 0.4430
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8846 - loss: 0.2748
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 3ms/step - accuracy: 0.9007 - loss: 0.2409
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9090 - loss: 0.2243
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9212 - loss: 0.1937
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9311 - loss: 0.1741
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9379 - loss: 0.1649
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9422 - loss: 0.1582
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9473 - loss: 0.1398
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 3ms/step - accuracy: 0.9482 - loss: 0.1433
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9514 - loss: 0.1163
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7944 - loss: 0.4278
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.8733 - loss: 0.2854
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9020 - loss: 0.2367
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9196 - loss: 0.2009
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9344 - loss: 0.1777
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9343 - loss: 0.1656
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9461 - loss: 0.1383
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9484 - loss: 0.1310
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9534 - loss: 0.1212
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9543 - loss: 0.1100
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9578 - loss: 0.1061
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7940 - loss: 0.4468
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8751 - loss: 0.2972
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.8982 - loss: 0.2533
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9071 - loss: 0.2187
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9280 - loss: 0.1904
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9393 - loss: 0.1869
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9314 - loss: 0.1700
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9460 - loss: 0.1401
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9446 - loss: 0.1419
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9543 - loss: 0.1236
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9582 - loss: 0.1178
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.7972 - loss: 0.4304
Epoch 2/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.8840 - loss: 0.2757
Epoch 3/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9024 - loss: 0.2457
Epoch 4/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9194 - loss: 0.1950
Epoch 5/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 2s 2ms/step - accuracy: 0.9281 - loss: 0.1842
Epoch 6/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9321 - loss: 0.1671
Epoch 7/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9526 - loss: 0.1350
Epoch 8/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9448 - loss: 0.1369
Epoch 9/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9606 - loss: 0.1149
Epoch 10/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - accuracy: 0.9564 - loss: 0.1108
Epoch 11/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy: 0.9571 - loss: 0.1122
Epoch 12/20
720/720 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step - accuracy

The accuracy of the second model in comparison to the first is instead poor, coming in at **93.83%** with 0.99% standard deviation after an additional hidden layer of 1 layer with 120 neurons, for a total of 2 hidden layers. Standardization was applied by using a standard scaler for our dataset. This represents only a 0.23% increase over the results of the first model. This suggests that simply adding more hidden layers to the model does not ensure improved performance; while it performs well, there isn't much of a performance difference between the modifications.

# Regression Problem

In [ ]:
import numpy
import pandas
from keras.models import Sequential
from keras.layers import Dense
from scikeras.wrappers import KerasRegressor
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import Pipeline

In [ ]:

columns = ['bedrooms', 'bathrooms', 'sqft_living', 'sqft_lot', 'floors',
           'waterfront', 'view', 'condition', 'grade', 'sqft_above',
           'sqft_basement', 'sqft_living15', 'sqft_lot15']
house_x = housingDF[list(columns)].astype('object').values
house_y = housingDF["price"].astype('object').values

In [ ]:
# define base mode
def baseline_model():
  # create model
  model = Sequential()
  model.add(Dense(18, input_shape=(len(columns),), kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(1, kernel_initializer='normal'))
  # Compile model
  model.compile(loss= 'mean_squared_error' , optimizer= 'adam')
  return model

In [ ]:
# fix random seed for reproducibility
seed = 7
numpy.random.seed(seed)
# evaluate model with standardized dataset
estimator = KerasRegressor(build_fn=baseline_model, epochs=10, batch_size=10, verbose=1)
kfold = KFold(n_splits=10)
results = cross_val_score(estimator, house_x, house_y, cv=kfold, scoring='neg_mean_squared_error')
print("Baseline: %.2f (%.2f) MSE" % (results.mean(), results.std()))

Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 244846034944.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 131561922560.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 106988339200.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 98966290432.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 89014083584.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 92194054144.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 75612004352.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 7s 3ms/step - loss: 68893368320.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 67692236800.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 65071828992.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 239941451776.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 126005854208.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 104925904896.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 99161587712.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 87263535104.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 76772139008.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 71267917824.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 67259052032.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 65841688576.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 64128745472.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 259225911296.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 127204212736.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 1ms/step - loss: 120737964032.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 106785308672.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 90001154048.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 77660758016.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 70740189184.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 70639099904.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 67759968256.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 68210155520.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 255959416832.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 1ms/step - loss: 123142627328.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 109181476864.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 107317567488.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 85535260672.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 79636586496.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 74194264064.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 69940748288.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 67962515456.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 65669849088.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 261320966144.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 1ms/step - loss: 121555533824.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 108831424512.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 94762754048.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 85950799872.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 81878458368.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 70979051520.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 71584227328.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 72482037760.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 68939128832.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 265538699264.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 110400258048.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 110034599936.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 86607675392.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 78475272192.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 73128198144.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 67005108224.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 1ms/step - loss: 69640814592.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 69391613952.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 66654003200.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 80s 2ms/step - loss: 252663857152.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 124031385600.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 112063111168.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 1ms/step - loss: 102129115136.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 86521339904.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 86072410112.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 70397845504.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 65017208832.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 69950586880.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 67300319232.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 260644257792.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 123021082624.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 118591741952.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 93015056384.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 84821516288.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 76647235584.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 73985269760.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 68070391808.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 64299532288.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 69069225984.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 238135951360.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 124093693952.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 114505359360.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 97462861824.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 93535330304.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 84425113600.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 77918437376.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 65351573504.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 69882732544.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 1ms/step - loss: 63789645824.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 1ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 261580587008.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 120923054080.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 113455259648.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 106205765632.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 94602600448.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 1ms/step - loss: 82722988032.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 72914477056.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 65420709888.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 69525569536.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 61945569280.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 0s 2ms/step
Baseline: -66210922141.56 (8581226588.35) MSE


I have used the housing regressional type dataset for the initial model. I obtain a wildly incomprehensible Mean Squared Error of about -60 billion using a base layer of 18 and a single output layer, with little to no preprocessing or standardization. This could be caused by either the dataset itself or the lack of data preparation.

In [ ]:
def model_2():
  model = Sequential()
  model.add(Dense(60, input_shape=(len(columns),), kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(200, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(300, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(1, kernel_initializer='normal'))
  model.compile(loss= 'mean_squared_error' , optimizer= 'adam')
  return model
# evaluate model with standardized dataset
estimators = []
estimators.append(('standardize', StandardScaler()))
estimators.append(('mlp', KerasRegressor(build_fn=model_2, epochs=10, batch_size=10, verbose=1)))
pipeline = Pipeline(estimators)
kfold = KFold(n_splits=10)
results = cross_val_score(pipeline, house_x, house_y, cv=kfold, scoring='neg_mean_squared_error')
print("Baseline: %.2f (%.2f) MSE" % (results.mean(), results.std()))

Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 224528318464.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 52706959360.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 48041660416.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 49080061952.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 44591718400.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 44861181952.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 42689581056.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 40727760896.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 41897197568.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 41210294272.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 212769783808.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44527665152.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 49221373952.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 45478838272.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 47079985152.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 43924852736.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 43933032448.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 41349300224.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 41309687808.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 40057143296.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 207991357440.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 47988826112.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 47433441280.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 46674055168.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 46372663296.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 49023852544.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 44532109312.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 46286643200.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 43208765440.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 43620749312.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 212692926464.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 47504269312.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 44948668416.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 48684060672.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 46419193856.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 43623858176.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44876156928.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 45148069888.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 41798496256.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 40830537728.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 199696007168.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 48680394752.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 49823367168.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 46514225152.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 45264777216.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 44656033792.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 43544899584.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44855078912.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 40469852160.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 40289148928.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 218150731776.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 50317402112.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 46774939648.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 46425489408.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 46547116032.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 43948658688.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44620746752.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 43802447872.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 40796897280.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 44218667008.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 218133987328.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 49173995520.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 46975844352.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 48220475392.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 42658529280.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 46664314880.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 45874917376.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 48865017856.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 39844655104.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 41096790016.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 21s 2ms/step - loss: 212664287232.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 50091692032.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 47503265792.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44074348544.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 45245382656.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 43036446720.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44400824320.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44488671232.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 43811721216.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 41569730560.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 201169731584.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 8s 2ms/step - loss: 49541869568.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 49927815168.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 48401956864.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 45990080512.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 46072664064.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 44164644864.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 42974302208.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 46326272000.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 42098016256.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 212747665408.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 48194519040.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 46613078016.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 46857220096.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 43885338624.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 43202588672.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 2ms/step - loss: 40420167680.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 2ms/step - loss: 45195304960.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 4s 2ms/step - loss: 42645839872.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 3s 2ms/step - loss: 41312399360.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step
Baseline: -43613200234.26 (3491511238.88) MSE


The second model has two additional hidden layers with 200 and 400 nodes each, in addition to standardization. The resulting MSE is still in the billions, but it varies by 22 billion from the first model. While this may seem significant, there is still a long way to go before the model can even be considered to perform well.

In [ ]:
# Normalize X values
from sklearn import preprocessing as prep
min_max_scaler = prep.MinMaxScaler()
house_x = min_max_scaler.fit_transform(house_x)

In [ ]:
house_x

array([[0.06060606, 0.125     , 0.06716981, ..., 0.        , 0.16193426,
        0.00574235],
       [0.09090909, 0.28125   , 0.17207547, ..., 0.08298755, 0.22216486,
        0.00802712],
       [0.06060606, 0.125     , 0.03622642, ..., 0.        , 0.3994149 ,
        0.00851302],
       ...,
       [0.06060606, 0.09375   , 0.05509434, ..., 0.        , 0.10686629,
        0.00155764],
       [0.09090909, 0.3125    , 0.09886792, ..., 0.        , 0.17398038,
        0.00073057],
       [0.06060606, 0.09375   , 0.05509434, ..., 0.        , 0.10686629,
        0.00081098]])

In [ ]:
def model_3():
  model = Sequential()
  model.add(Dense(60, input_shape=(len(columns),), kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(512, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(1024, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(1024, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(1024, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(512, kernel_initializer='normal', activation= 'relu'))
  model.add(Dense(1, kernel_initializer='normal'))
  model.compile(loss= 'mean_squared_error' , optimizer= 'adam')
  return model
# evaluate model with standardized dataset
estimators = []
estimators.append(('standardize', StandardScaler()))
estimators.append(('mlp', KerasRegressor(build_fn=model_3, epochs=10, batch_size=10, verbose=1)))
pipeline = Pipeline(estimators)
kfold = KFold(n_splits=10)
results = cross_val_score(pipeline, house_x, house_y, cv=kfold, scoring='neg_mean_squared_error')
print("Model 3: %.2f (%.2f) MSE" % (results.mean(), results.std()))

/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


Epoch 1/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 4ms/step - loss: 79457222656.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 44547874816.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 3ms/step - loss: 41790558208.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42757201920.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 43046768640.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 42229415936.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 41755627520.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 41535381504.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 37956366336.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 3ms/step - loss: 38777581568.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 4ms/step - loss: 80098787328.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 8s 3ms/step - loss: 48576585728.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 41718734848.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 40008380416.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 39062286336.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 41199255552.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 39545905152.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38338555904.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38706081792.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 3ms/step - loss: 38705668096.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 4ms/step - loss: 87602618368.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 50308284416.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 42772688896.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42018971648.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42390495232.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42836340736.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38966165504.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42209603584.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 7s 4ms/step - loss: 38601134080.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 37897555968.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 4ms/step - loss: 79027200000.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 5s 3ms/step - loss: 45649362944.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 45605920768.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 3ms/step - loss: 45055397888.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 39003410432.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 7s 3ms/step - loss: 40992419840.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 41493598208.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 38378287104.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 38339317760.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 39953137664.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 9s 4ms/step - loss: 93119348736.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 46587199488.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 46674853888.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 47316164608.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 43446231040.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 44552298496.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42711425024.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38690004992.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 38759313408.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 40347746304.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 9s 3ms/step - loss: 84845592576.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 49273606144.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 46278664192.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 49688965120.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 42879852544.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 45232971776.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 41253855232.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 44392263680.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38048772096.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 4ms/step - loss: 40199311360.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 9s 3ms/step - loss: 87678369792.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 7s 3ms/step - loss: 53773574144.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 44836835328.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 44344340480.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 3ms/step - loss: 46134779904.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 44277788672.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 46837395456.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 44924301312.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 40821559296.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38643298304.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 4ms/step - loss: 92315033600.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 46394327040.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 48696606720.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 45002313728.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 42632921088.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 41343619072.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 41775706112.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 13s 7ms/step - loss: 40639508480.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38230462464.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 42759270400.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 4ms/step - loss: 86162628608.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 46432157696.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 50876186624.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 45812998144.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 47719165952.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 39855280128.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 39323860992.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 39656689664.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38744297472.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 38721003520.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 2ms/step
Epoch 1/10


/usr/local/lib/python3.10/dist-packages/scikeras/wrappers.py:925: UserWarning: ``build_fn`` will be renamed to ``model`` in a future release, at which point use of ``build_fn`` will raise an Error instead.
  X, y = self._initialize(X, y)
/usr/local/lib/python3.10/dist-packages/keras/src/layers/core/dense.py:86: UserWarning: Do not pass an `input_shape`/`input_dim` argument to a layer. When using Sequential models, prefer using an `Input(shape)` object as the first layer in the model instead.
  super().__init__(activity_regularizer=activity_regularizer, **kwargs)


1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 4ms/step - loss: 84099104768.0000
Epoch 2/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 48239042560.0000
Epoch 3/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 45241778176.0000
Epoch 4/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 10s 3ms/step - loss: 39696142336.0000
Epoch 5/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 7s 3ms/step - loss: 43756605440.0000
Epoch 6/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 40070828032.0000
Epoch 7/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 7s 3ms/step - loss: 39676039168.0000
Epoch 8/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 43082403840.0000
Epoch 9/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 11s 3ms/step - loss: 39387553792.0000
Epoch 10/10
1946/1946 ━━━━━━━━━━━━━━━━━━━━ 6s 3ms/step - loss: 37244801024.0000
217/217 ━━━━━━━━━━━━━━━━━━━━ 1s 3ms/step
Model 3: -40236050696.61 (4261737963.39) MSE


In [ ]:
print("Model 1: %.3f (%.3f) RMSE" % (np.sqrt(66210922141.56), np.sqrt((8581226588.35))))
print("Model 2: %.3f (%.3f) RMSE" % (np.sqrt(43613200234.26), np.sqrt((3491511238.88))))
print("Model 3: %.3f (%.3f) RMSE" % (np.sqrt(-(results.mean())), np.sqrt((results.std()))))


Model 1: 257314.831 (92634.910) RMSE
Model 2: 208837.737 (59089.011) RMSE
Model 3: 200589.259 (65281.988) RMSE


The evaluation results are easier to understand when the root mean squared error is used instead of our MSE, which has billions. Based on our interpretation of the data, the model's predicted values differ by approximately 200k from the corresponding actual values, indicating that they are far from perfect. On the other hand, Model 2 employs standardization and has a larger network, while Model 3 uses the input data that has been normalized before it is standardized and has an even larger network. Model 1 is our baseline model, with a base amount of layers and little to no preprocessing. It is evident that there is a significant difference in the error score between Models 1 and 2, but not between Models 2 and 3.